scripts/frontmatter/get-frontmatter.js

```diff
@@ -0,0 +1,93 @@
+const frontmatter = require('@github-docs/frontmatter');
+const { TYPES } = require('../constants');
+
+const GATSBY_CONTENT_TYPES = {
+  [TYPES.BASIC_PAGE]: 'page',
+  [TYPES.API_DOC]: 'apiDoc',
+  [TYPES.RELEASE_NOTE]: 'releaseNote',
+  [TYPES.RELEASE_NOTE_PLATFORM]: 'releaseNotePlatform',
+  [TYPES.TROUBLESHOOTING]: 'troubleshootingDoc',
+  [TYPES.WHATS_NEW]: 'nr1Announcement',
+};
+
+const GATSBY_TEMPLATE = {
+  [TYPES.BASIC_PAGE]: 'basicDoc',
+  [TYPES.API_DOC]: 'basicDoc',
+  [TYPES.RELEASE_NOTE]: 'basicDoc',
+  [TYPES.RELEASE_NOTE_PLATFORM]: 'basicDoc',
+  [TYPES.TROUBLESHOOTING]: 'basicDoc',
+  [TYPES.WHATS_NEW]: 'basicDoc',
+};
+
+const FRONTMATTER_TEMPLATE = {
+  [TYPES.BASIC_PAGE]: 'page',
+  [TYPES.API_DOC]: 'apiDoc',
+  [TYPES.RELEASE_NOTE]: 'releaseNote',
+  [TYPES.RELEASE_NOTE_PLATFORM]: 'releaseNotePlatform',
+  [TYPES.TROUBLESHOOTING]: 'troubleshooting',
+  [TYPES.WHATS_NEW]: 'whatsNew',
+};
+
+const getFrontmatter = (type, doc) => {
+  const defaultFrontmatter = {
+    title: doc.title.replace(':', '-'),
+    contentType: GATSBY_CONTENT_TYPES[type],
+    template: GATSBY_TEMPLATE[type],
+    topics: getTopics(doc),
+  };
+
+  return frontmatter.stringify('', {
+    ...defaultFrontmatter,
+    ...addCustomFrontmatter[FRONTMATTER_TEMPLATE[type]](doc),
+  });
+};
+
+const addCustomFrontmatter = {
+  apiDoc: (doc) => {
+    return;
+  },
+  page: (doc) => {
+    return {
+      japaneseVersion: doc.japaneseVersionExists === 'yes',
+    };
+  },
+  troubleshooting: (doc) => {
+    return {
+      japaneseVersion: doc.japaneseVersionExists === 'yes',
+    };
+  },
+  releaseNote: (doc) => {
+    return {
+      releaseDateTime: doc.releasedOn || '',
+      releaseVersion: doc.releaseVersion || '',
+      downloadLink: doc.downloadLink || '',
+    };
+  },
+  releaseNotePlatform: (doc) => {
+    return {
+      releaseDateTime: doc.releasedOn || '',
+      releaseImpact: doc.releaseImpact || [],
+      downloadLink: doc.downloadLink || '',
+    };
+  },
+  whatsNew: (doc) => {
+    return {
+      summary: doc.summary || '',
+      learnMoreLink: doc.learnMoreLink || '',
+      getStartedLink: doc.getStartedLink || '',
+    };
+  },
+};
+
+const getTopics = (doc) => {
+  const topics = [];
+  for (let i = 1; i < 4; i++) {
```

Comment:
Could you talk through why you use `4` here? Seems like a magic number to me. Is there a better way to get this?

I see that you're pulling a topic by its key prefixed with `topic_<i>`. Would it be better to filter topics by the prefix instead?

```js
const topics = Object.entries(doc).reduce((topics, [key, value]) => 
  key.startsWith('topic_') ? [...topics, value] : topics
), [])
```

This might be a bit more resilient in case there are more than 4 topics in any give document (unlikely, but handles it correctly if so)

Reply:
So there are only three topics that are returned from the JSON endpoint, which is why I was using that number. I agree that the way you suggested is more resilient and I will make that change!